libs/datasets/timeseries.py

```diff
@@ -1098,29 +1119,19 @@ def _aggregate_dataframe_by_region(
     long_all_values = df.rename_axis(columns=PdFields.VARIABLE).stack(dropna=True)
     assert long_all_values.index.names == [CommonFields.LOCATION_ID] + groupby_columns
 
-    # Make a Series with index `location_id_agg` containing the count of regions to be
-    # aggregated for that region.
-    location_id_agg_count = (
-        pd.Series(list(location_id_map.values()))
-        .value_counts()
-        .rename_axis(LOCATION_ID_AGG)
-        .rename("location_id_agg_count")
-    )
-
-    # Aggregate by location_id_agg, optional date and variable. Keep the sum and count of
-    # input values.
-    long_agg = long_all_values.groupby(groupby_columns, sort=False).agg(["sum", "count"])
-    # Join the count of regions in each location_id_agg
-    long_agg = long_agg.join(location_id_agg_count, on=LOCATION_ID_AGG)
+    # Aggregate by location_id_agg, optional date and variable.
+    long_agg = long_all_values.groupby(groupby_columns, sort=False).sum()
 
-    if not ignore_na:
-        # Only keep aggregated values where the count of aggregated values is the same as the
-        # count of input regions.
-        long_agg = long_agg.loc[long_agg["count"] == long_agg["location_id_agg_count"]]
+    if reporting_ratio_required:
+        weighted_reporting_ratio = _calculate_weighted_reporting_ratio(
+            long_all_values, location_id_map, reporting_ratio_location_weights, groupby_columns
+        )
+        is_valid_reporting_ratio = weighted_reporting_ratio >= reporting_ratio_required
+        acceptable_ratios = weighted_reporting_ratio.loc[is_valid_reporting_ratio]
```

Comment:
not sure I'm following all the index but not sure you need acceptable_ratios if is_valid_reporting_ratio is a mask of locations with good ratios. How about

long_agg = long_agg.loc[is_valid_reporting_ratio]